File: bpm_mobile_client/lib/core/models/sync_request_dto.dart
class SyncRequestDTO {
  final String politicaId;
  final String usuarioId;
  final String? clienteId;
  final String? documentoCliente;
  final String? clienteNombre;
  final String? offlineId;

  SyncRequestDTO({
    required this.politicaId,
    required this.usuarioId,
    this.clienteId,
    this.documentoCliente,
    this.clienteNombre,
    this.offlineId,
  });

  Map<String, dynamic> toJson() {
    return {
      'politicaId': politicaId,
      'usuarioId': usuarioId,
      'clienteId': clienteId,
      'documentoCliente': documentoCliente,
      'clienteNombre': clienteNombre,
      'offlineId': offlineId,
    };
  }
}

File: backend-core/src/main/java/com/bpm/inteligente/dto/SyncRequestDTO.java
package com.bpm.inteligente.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class SyncRequestDTO implements Serializable {

    @NotBlank(message = "El politicaId es obligatorio")
    private String politicaId;

    @NotBlank(message = "El usuarioId es obligatorio")
    private String usuarioId;

    private String clienteId;
    private String documentoCliente;
    private String clienteNombre;
    private String offlineId;
}
